src/server/pfs/s3/error.go

```diff
@@ -0,0 +1,105 @@
+package s3
+
+import (
+	"net/http"
+	"regexp"
+)
+
+// note: the repo matcher is stricter than what PFS enforces, because this is
+// based off of minio's bucket matching regexp. At the point where we're
+// matching errors, we shouldn't be attempting to operate on buckets that
+// don't match the minio regexp anyways.
+var (
+	repoNotFoundMatcher   = regexp.MustCompile(`repos/ ?[a-zA-Z0-9.\-_]{1,255} not found`)
+	branchNotFoundMatcher = regexp.MustCompile(`branches/[a-zA-Z0-9.\-_]{1,255}/ [^ ]+ not found`)
+	fileNotFoundMatcher   = regexp.MustCompile(`file .+ not found`)
+)
```

Comment:
These would be better placed in `src/server/pfs/pfs.go` since they may well be of use elsewhere, and we want to minimize error check fragmentation as much as possible. They should also be wrapped in `IsXErr` functions.

Reply:
should I update `errutil.IsNotFoundError` to use these new functions as well?

Reply:
Yeah, assuming it keeps the same semantics and makes things more unified / cleaner I'm all for it. Be careful changing `errutil.IsNotFoundError`, it's used in a bunch of places and if its semantics change it's possible to introduce all sorts of crazy bugs. Error matching is easily the most fragile part of go (or at least our usage of go) so I'm always really paranoid when changing it.

Reply:
I won't change it then - I'm not sure if these regexes cover all the possible ways things could not be found in PFS